pkg/resource/deploy/source_eval.go

```diff
@@ -1368,6 +1394,10 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 	if err != nil {
 		return nil, err
 	}
+
+	// TODO determine if should be a reference.
+	rm.resGoals[urn] = *goal
```

Comment:
could also be moved earlier potentially or run as a defer.

Reply:
I would just add it to the bottom of the else clause of the `if remote` check. Something like (with lock):

```go
if result != nil && result.State != nil && result.State.URN != "" {
	func() {
		rm.resGoalsLock.Lock()
		defer rm.resGoalsLock.Unlock()
		rm.resGoals[result.State.URN] = *goal
	}()
}
```